File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/detectProperties/MergedPropertyReceiverShadowedByParam.java
// IGNORE_K2
interface Container {
    Item getItem();
}

class Item {
    int indexOf(Object o) {
        return 0;
    }
}

class ContainerImpl implements Container {
    private final Item mItem;

    ContainerImpl(Item item) {
        mItem = item;
    }

    @Override
    public Item getItem() {
        return mItem;
    }

    public int locate(Object item) {
        return mItem.indexOf(item);
    }
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/detectProperties/MergedPropertyReceiverShadowedByParam.kt
internal interface Container {
    val item: Item
}

internal class Item {
    fun indexOf(o: Any?): Int {
        return 0
    }
}

internal class ContainerImpl(override val item: Item) : Container {
    fun locate(item: Any?): Int {
        return this.item.indexOf(item)
    }
}
